p2p/peer.go

```diff
@@ -181,16 +184,22 @@ func (p *Peer) Validator() bool {
 	return p.rw.is(validatorConn)
 }
 
-func newPeer(conn *conn, protocols []Protocol) *Peer {
+// Sentry returns true if the peer is a sentry connection
+func (p *Peer) Sentry() bool {
+	return p.rw.is(sentryConn)
+}
+
+func newPeer(conn *conn, protocols []Protocol, IsProxied bool) *Peer {
```

Comment:
Instead of an `IsProxied` parameter, pass in a `p2p server` parameter.  And have the `eth` layer check to see if it's coming from the proxy server.